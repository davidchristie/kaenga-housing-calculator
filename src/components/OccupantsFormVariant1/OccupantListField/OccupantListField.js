import React, { Component, PropTypes } from 'react'
import { Button, ListGroup } from 'react-bootstrap'

import OccupantField from './OccupantField'

class OccupantFields extends Component {
  render () {
    const { fields } = this.props
    return (
      <div>
        <Button onClick={() => fields.push({})}>
          Add Occupant
        </Button>
        <ListGroup>
          {fields.map((occupant, index) =>
            <OccupantField
              index={index}
              key={index}
              occupant={occupant}
              remove={() => fields.remove(index)}
            />
          )}
        </ListGroup>
      </div>
    )
  }
}

OccupantFields.defaultProps = {
  fields: []
}

OccupantFields.propTypes = {
  fields: PropTypes.array.isRequired
}

export default OccupantFields
