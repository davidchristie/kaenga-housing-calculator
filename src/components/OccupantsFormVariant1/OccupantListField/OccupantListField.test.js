import React from 'react'
import ReactTestUtils from 'react-addons-test-utils'

import OccupantListField from './OccupantListField'

const shallowRenderer = ReactTestUtils.createRenderer()

it('matches snapshot', () => {
  expect(
    shallowRenderer.render(<OccupantListField />)
  ).toMatchSnapshot()
})
