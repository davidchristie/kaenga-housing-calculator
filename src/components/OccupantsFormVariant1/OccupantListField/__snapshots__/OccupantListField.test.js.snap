exports[`test matches snapshot 1`] = `
<div>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}>
    Add Occupant
  </Button>
  <ListGroup
    bsClass="list-group">
    
  </ListGroup>
</div>
`;
